### DISCUSSION TITLE
ARM support

### DISCUSSION THREAD

**Discussion by otto76** [taligentx/dscKeybusInterface#124] [dscKeybusInterface Discussion]
Hi, guys, do we have plans to support ARM boards, aka Arduino Due?
Should be some challenges with interrupt handling. 

But in general, is there anything else, why its put a side for now? 
O.

**Comment by taligentx** [comment]
Hi @otto76 - late response but thanks for the suggestion, is there a particular advantage of the Due vs the ATmega-based Arduino boards in this application?

I like supporting the Uno, etc as the boards are commonly available at local electronics shops worldwide, but the $3-$5USD esp8266/esp32 boards with integrated WiFi are more capable than the $40 Due board.

**Comment by otto76** [comment]
Hey, I prefer wired solution instead for wireless for security stuff, therefore dont like esp<xx> board. 
Regular Uno, has very limited amount of memory available

As I have several other monitoring activities going on for the same board, then I am running out of memory, plus performance is lagging. I do have DUE on my hands, thus was interested for solution 

I tried to rewrite code for timers and other DUE specific stuff, but w/ very little success 
Therefore dropped question here :)
o.

**Comment by Dilbert66** [comment]
esp32 does support ethernet .
https://www.olimex.com/Products/IoT/ESP32/ESP32-POE/open-source-hardware

**Comment by taligentx** [comment]
Certainly understandable if you have the Due hardware on hand - if you or anyone else gets the interrupts and timers adjusted for the Due, I can merge a PR but it's not currently on my roadmap (if nothing else, I don't have a Due to test with).  

The Arduino Uno, esp8266 and esp32 can connect to the ENC28J60, W5100, and W5500 Ethernet modules over SPI so that's one way to go for a wired connection.

As for performance: https://forum.arduino.cc/index.php?topic=337653.0

The Due is great compared to the atmega328p, but the esp8266@160MHz and esp32 exceed the Due's performance, SRAM, and flash memory specs at 1/10th the cost.

I'll leave the issue open if someone is able to adjust the code for the Due.
